lib/msal-node/test/cache/Storage.spec.ts

```diff
@@ -0,0 +1,161 @@
+
+        const loggerOptions = {
+            loggerCallback: (
+                level: LogLevel,
+                message: string,
+                containsPii: boolean
+            ) => {
+                debug(`msal:${LogLevel[level]}${containsPii ? '-Pii' : ''}`)(message);
+            },
+            piiLoggingEnabled: false,
+            logLevel: LogLevel.Info,
+        };
+        logger = new Logger(loggerOptions!);
+    });
+
+    it("Constructor tests: ", () => {
+        const nodeStorage = new Storage(logger);
+        expect(nodeStorage).toBeInstanceOf(Storage);
+
+        const cache = nodeStorage.getCache();
+        expect(Object.keys(cache).length).toBe(0);
+
+        const inMemoryCache = nodeStorage.getInMemoryCache();
+        expect(Object.keys(inMemoryCache.accessTokens).length).toBe(0);
+    });
+
+    it("setInMemoryCache() and getCache()", () => {
+        const nodeStorage = new Storage(logger);
+        nodeStorage.setInMemoryCache(inMemoryCache);
+
+        const cache = nodeStorage.getCache();
+        const accountKey = 'uid.utid-login.microsoftonline.com-microsoft';
+        const account = cache[accountKey] as AccountEntity;
+        expect(account).toBeInstanceOf(AccountEntity);
+        expect(account.clientInfo).toBe('eyJ1aWQiOiJ1aWQiLCAidXRpZCI6InV0aWQifQ==');
+    });
+
+    it('setInMemoryCache() and getInMemoryCache() tests', () => {
+        const nodeStorage = new Storage(logger);
+        nodeStorage.setInMemoryCache(inMemoryCache);
+
+        const cache = nodeStorage.getCache();
+        const accountKey = 'uid.utid-login.microsoftonline.com-microsoft';
+        const account = cache[accountKey] as AccountEntity;
+        expect(account).toBeInstanceOf(AccountEntity);
+        expect(account.clientInfo).toBe(
+            'eyJ1aWQiOiJ1aWQiLCAidXRpZCI6InV0aWQifQ=='
+        );
+
+        const newInMemoryCache = nodeStorage.getInMemoryCache();
+        expect(newInMemoryCache.accounts[accountKey]).toEqual(account);
+    });
+
+    it('setItem() and getItem() tests', () => {
+        const nodeStorage = new Storage(logger);
+        nodeStorage.setInMemoryCache(inMemoryCache);
+
+        const accountKey = 'uid1.utid1-login.windows.net-samplerealm';
+        const newMockAccount = {
+            'uid1.utid1-login.windows.net-samplerealm': {
+                username: 'Jane Doe',
+                local_account_id: 'object5678',
+                realm: 'samplerealm',
+                environment: 'login.windows.net',
+                home_account_id: 'uid1.utid1',
+                authority_type: 'MSSTS',
+                client_info: 'eyJ1aWQiOiJ1aWQxIiwgInV0aWQiOiJ1dGlkMSJ9',
+            },
+        };
+        let account = new AccountEntity();
+        account = CacheManager.toObject(account, newMockAccount);
+
+        nodeStorage.setItem(accountKey, account);
+        const fetchedAccount = nodeStorage.getItem(accountKey);
+
+        expect(account).toEqual(fetchedAccount);
```

Comment:
I think it would be good to also test instanceOf here as well since we're expecting getItem to return specific types

Reply:
Added